aten/src/ATen/native/cpu/CatKernel.cpp

```diff
@@ -18,29 +17,27 @@ struct InputMeta {
     , inner_size(t.size(dim) * inner) {}
 };
 
+namespace {
+
 template <typename scalar_t>
-void cat_serial_kernel_impl(Tensor& result, TensorList tensors, int64_t dim) {
-  auto size = result.sizes().vec();
-  int64_t outer = 1, inner = 1;
-  for (int64_t i = 0; i < dim; i++) {
-    outer *= size[i];
-  }
-  for (int64_t i = dim + 1; i < size.size(); i++) {
-    inner *= size[i];
-  }
+void cat_contig_kernel_impl(Tensor& result, TensorList tensors, int64_t dim) {
+  int64_t outer = result.numel() / (result.size(dim) * result.stride(dim));
   scalar_t* result_data = result.data_ptr<scalar_t>();
+  // Adjust grain size to account for the work done
```

Comment:
nit: do you need this comments in the code base

Reply:
Yes. No need for it here.